### PR TITLE
RavenDB-15589 Implement aggressive caching on lazy queries

### PR DESCRIPTION
**Pull request by ayende**

### CODE REVIEW

File: src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs [hunk 1]
```diff
@@ -13,21 +13,55 @@ namespace Raven.Client.Documents.Commands.MultiGet
...
             _baseUrl = $"{node.Url}/databases/{node.Database}";
+            url = $"{_baseUrl}/multi_get";
+
+            var aggressiveCacheOptions = _requestExecutor.AggressiveCaching.Value;
+            if (aggressiveCacheOptions?.Mode == AggressiveCacheMode.TrackChanges)
+            {
+                Result = new List<GetResponse>();
+                foreach (var command in _commands)
+                {
+                    if(command.SkipAggressiveCache) break;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs]: Should be `continue` ?

File: src/Raven.Client/Documents/Commands/MultiGet/GetRequest.cs [hunk 2]
```diff
@@ -22,6 +22,8 @@ public class GetRequest
         /// </summary>
         public string Query { get; set; }
         
+        public bool SkipAggressiveCache { get; set; }
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Commands/MultiGet/GetRequest.cs]: It is called 'CanCacheAggressively' in Commands. Let's stick to that.

File: src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs [hunk 3]
```diff
@@ -13,21 +13,55 @@ namespace Raven.Client.Documents.Commands.MultiGet
...
+                    var cacheKey = GetCacheKey(command, out string _);
+                    using (var cachedItem = _cache.Get(ctx, cacheKey, out _, out var cached))
+                    {
+                        if (cached == null || 
+                            cachedItem.Age > aggressiveCacheOptions.Duration ||
+                            cachedItem.MightHaveBeenModified)
+                            break;
+
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs:49]: Double-check. Do we need to apply logic here from 'RequestExecutor.TryGetFromCache' ?

File: src/Raven.Client/Documents/Commands/MultiGet/GetRequest.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.Client/Documents/Commands/MultiGet/GetRequest.cs]: I wanted to have this set to false by default, so we don't have to touch it in all call sites.

File: src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs]: No, the first one that isn't find for us, we'll break and the `Result`'s count will be different than `_commands` so we'll go to the server for all items.

File: src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs (hunk 3 above)
Comment (ayende) [reply src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs:49]: That is where I took the logic from, yes.

File: src/Raven.Client/Documents/Commands/MultiGet/GetRequest.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Commands/MultiGet/GetRequest.cs]: Set it to `true` in ctor like we have it for RavenCommand?

File: src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs]: OK

File: src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs (hunk 3 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs:49]: There was additional logic for delete, but it is not possible to have that case here, wanted to double-check.

**Review by ppekrol — APPROVED** [review]

**Pull request merged**